servicetalk-client-api/src/main/java/io/servicetalk/client/api/ServiceDiscoverer.java

```diff
@@ -47,5 +49,5 @@
      * </ul>
      * @return a {@link Publisher} that represents a stream of events from the service discovery system.
      */
-    Publisher<E> discover(UnresolvedAddress address);
+    Publisher<List<E>> discover(UnresolvedAddress address);
```

Comment:
Do we need a `List`? Will `Collection` be enough?

Reply:
Good call, `Collection` is enough